### PR TITLE
[USETUP] Display the chosen filesystem in case of bootcode installation failure

### PR DESCRIPTION
**Pull request by Fraizeraust**
## Purpose
Thanks to Victor Perevertkin, it is possible to boot ReactOS onto a BTRFS partition. However if for some weird reason the BTRFS bootcode fails to be installed within the partition, the error message will still display `Setup failed to install FAT bootcode on the system partition.` which the user may think the USETUP attempted to install the FAT bootcode instead of BTRFS, which is very misleading.
## Presentation
**URL Image:** https://imgur.com/a/iqNKzlQ
## Proposed Changes
- Rename ERROR_INSTALL_BOOTCODE and ERROR_WRITE_BOOT to ERROR_INSTALL_BOOTCODE_FAT/ERROR_WRITE_BOOT_FAT
- Add ERROR_INSTALL_BOOTCODE_BTRFS and ERROR_WRITE_BOOT_BTRFS
- Include the error codes both in the ReactOS USETUP errorcode.h header and the setup library
- Implement two message descriptions for ERROR_WRITE_BOOT_BTRFS/ERROR_INSTALL_BOOTCODE_BTRFS in resource headers
- In usetup.c, check the chosen filesystem for the partition and display it accordingly in case of bootcode installation failure

### CODE REVIEW

File: base/setup/usetup/usetup.c [hunk 1]
```diff
@@ -4461,8 +4476,23 @@ BootLoaderHarddiskMbrPage(PINPUT_RECORD Ir)
                                    PartitionType);
     if (!NT_SUCCESS(Status))
     {
-        MUIDisplayError(ERROR_WRITE_BOOT, Ir, POPUP_WAIT_ENTER);
-        return QUIT_PAGE;
+        /* Check the partition type, if we have FAT on it */
+        if (PartitionType == PARTITION_FAT32 ||
```
Comment (learn-more) [inline base/setup/usetup/usetup.c]: I would change this check to check `PARTITION_LINUX`, and just have an 'else' case with no condition for fat.
this makes it impossible to continue (as it was before)

**Comment by dsp2003** [comment]
Why not simply remember the partition type and use a single string for all possible filesystems?

**Comment by Extravert-ir** [comment]
Having a separate string for each filesystem doesn't look good for me.
Maybe just do a single string "Setup failed to install the bootloader code on the system partition"? I don't think that knowing which particular VBR failed here is worth it. Besides you can always guess it from your FS choice on previous steps

P.S. Btw why there are identical ERROR_INSTALL_BOOTCODE and ERROR_WRITE_BOOT?

**Comment by Fraizeraust** [comment]
> Having a separate string for each filesystem doesn't look good for me.

Care to explain objectively why is it bad showing the filesystem in the string?
> P.S. Btw why there are identical ERROR_INSTALL_BOOTCODE and ERROR_WRITE_BOOT?

Looking at the code, `ERROR_WRITE_BOOT` error message string is used as a parameter for `MUIDisplayError()` when `InstallVBRToPartition()` fails to install the VBR onto the partition meanwhile `ERROR_INSTALL_BOOTCODE` is used for `InstallMbrBootCodeToDisk()` when it fails. I cannot tell why the strings are identical, you may better ask Hermes on this.

**Comment by Extravert-ir** [comment]
> Care to explain objectively why is it bad showing the filesystem in the string?
With this approach we have a bunch of identical strings which are different only by an FS name in them - waste of time for translators. Moreover you will have to add another bunch when we add another FS to installer (or it will still show "FAT" string).

**Comment by HBelusca** [comment]
Superseded by commit 073c09e that implements the suggested improvements of Victor.
Thanks to both of you for having tried to solve this problem!

**Pull request closed without merging**